### PR TITLE
Remove initial docker unit

### LINKED ISSUE #1811: Docker daemon configuration disappears
**Is this a BUG REPORT or FEATURE REQUEST?**: BUG REPORT

**Minikube version**: v0.21.0

**Environment**:
- **OS**: Ubuntu 17.04
- **VM Driver**: virtualbox
- **ISO version**: minikube-v0.23.0.iso
- **Install tools**:
- **Others**:
I'm modyfing memory and Docker storage driver to use overlay2, but I'm not sure if it's related.
`minikube start --memory 4096 --docker-opt storage-driver=overlay2`

**What happened**:
After some time after starting minikube, the docker daemon inside dies, resulting in kubectl errors. 
Stopping and starting minikube solves the problem.

```
$ kubectl get pods
The connection to the server 192.168.99.100:8443 was refused - did you specify the right host or port?
```
I can SSH into minikube and then:
```
$ sudo systemctl status docker
[[0;1;31m●[[0m docker.service - Docker Application Container Engine
   Loaded: loaded (/lib/systemd/system/docker.service; enabled; vendor preset: enabled)
   Active: [[0;1;31mfailed[[0m (Result: exit-code) since Fri 2017-08-11 10:36:48 UTC; 21s ago
     Docs: https://docs.docker.com
  Process: 3599 ExecStart=/usr/bin/dockerd -H fd:// [[0;1;31m(code=exited, status=1/FAILURE)[[0m
 Main PID: 3599 (code=exited, status=1/FAILURE)
      CPU: 18ms

Aug 11 10:36:47 minikube systemd[1]: Starting Docker Application Container Engine...
Aug 11 10:36:47 minikube dockerd[3599]: time="2017-08-11T10:36:47.592883109Z" level=info msg="libcontainerd: new containerd process, pid: 3647"
Aug 11 10:36:48 minikube dockerd[3599]: time="2017-08-11T10:36:48.596286343Z" level=fatal msg="Error starting daemon: error initializing graphdriver: \"/mnt/sda1/var/lib/docker\" contains several valid gra
phdrivers: overlay2, overlay; Please cleanup or explicitly choose storage driver (-s <DRIVER>)"
Aug 11 10:36:48 minikube systemd[1]: [[0;1;39mdocker.service: Main process exited, code=exited, status=1/FAILURE[[0m
Aug 11 10:36:48 minikube systemd[1]: [[0;1;31mFailed to start Docker Application Container Engine.[[0m
Aug 11 10:36:48 minikube systemd[1]: [[0;1;39mdocker.service: Unit entered failed state.[[0m
Aug 11 10:36:48 minikube systemd[1]: [[0;1;39mdocker.service: Failed with result 'exit-code'.[[0m
```

```
$ sudo ls -al /mnt/sda1/var/lib/docker
total 120
drwx--x--x  11 root root  4096 Aug  8 11:21 .
drwxr-xr-x   9 root root  4096 Aug  8 11:21 ..
drwx------  66 root root 16384 Aug 11 08:18 containers
drwx------   4 root root  4096 Aug  8 11:21 image
drwxr-x---   3 root root  4096 Aug  8 11:21 network
drwx------   2 root root  4096 Aug  8 11:21 overlay
drwx------ 525 root root 65536 Aug 11 08:18 overlay2
drwx------   2 root root  4096 Aug  8 11:21 swarm
drwx------   2 root root  4096 Aug 10 10:16 tmp
drwx------   2 root root  4096 Aug  8 11:21 trust
drwx------  11 root root  4096 Aug 11 08:17 volumes
```

It seems it's the problem with storage driver, but really, the problem is that there's no 10-machine.conf drop-in for the systemd docker service.
Even if I didn't use overlay2, the daemon would fail due to missing certificates.

When minikube has just started:
```
$ ls -al /etc/systemd/system/docker.service.d/10-machine.conf      
-rw-r--r-- 1 root root 1814 Aug 11 08:17 /etc/systemd/system/docker.service.d/10-machine.conf

$ ls -al /etc/docker
total 16
drwxr-xr-x  2 root root    0 Aug 11 08:17 .
drwxr-xr-x 23 root root    0 Aug 11 08:17 ..
-rw-r--r--  1 root root    0 Jul 18 21:46 .keep
-rw-r--r--  1 root root 1034 Aug 11 08:17 ca.pem
-rw-------  1 root root  244 Aug 11 08:17 key.json
-rw-r--r--  1 root root 1675 Aug 11 08:17 server-key.pem
-rw-r--r--  1 root root 1107 Aug 11 08:17 server.pem
```

After the problem occurs:
```
$ ls -al /etc/systemd/system/
total 0
drwxr-xr-x 7 root root  0 Jul 18 21:46 .
drwxr-xr-x 5 root root  0 Jul 18 21:46 ..
lrwxrwxrwx 1 root root 33 Jul 18 22:05 ctrl-alt-del.target -> /lib/systemd/system/reboot.target
lrwxrwxrwx 1 root root 45 Jul 18 22:05 default.target -> ../../../lib/systemd/system/multi-user.target
drwxr-xr-x 2 root root  0 Jul 18 21:46 getty.target.wants
drwxr-xr-x 2 root root  0 Jul 18 22:23 multi-user.target.wants
drwxr-xr-x 2 root root  0 Jul 18 22:05 network-online.target.wants
drwxr-xr-x 2 root root  0 Jul 18 22:10 remote-fs.target.wants
drwxr-xr-x 2 root root  0 Jul 18 22:12 sockets.target.wants

$ ls -al /etc/docker
total 0
drwxr-xr-x  2 root root 0 Jul 18 21:46 .
drwxr-xr-x 22 root root 0 Aug 11 10:05 ..
-rw-r--r--  1 root root 0 Jul 18 21:46 .keep
```

I suspect that for some reason the VM could have restarted and the systemd Docker service is not configured properly upon reboot.

EDIT: the machine has restarted indeed. Is there any place I can check to see why?
```
$ uptime
 12:13:16 up 0 min,  1 user,  load average: 1.82, 0.67, 0.24
```

**What you expected to happen**:
Docker to continue running.

**How to reproduce it**:
Start minikube, wait for a bit, observe that Docker is dead.

**Output of `minikube logs`**:
`-- No entries --`
I've also tried to keep another shell with `minikube logs` running, but it didn't record anything interesting before the crash.

**Anything else do we need to know**:
Possibly might be the same as #1800 #1763.

### PR DESCRIPTION
**Pull request by r2d4**
Fixes #1811

This delete the default docker configuration in the minikube iso.  This should slightly decrease start time.  I ran into #1811 when caching images.

I've also cherry picked ff2667f, since that seems to fix all the flakiness we've been seeing with the integration tests.

Summaries from the commits:

The docker daemon initially starts up in the ISO, only to be stopped
and replaced by minikube's custom configuration.  This removes the
initial unit, so that docker doesn't automatically start up until
minikube restarts it with its configuration.

Rewrite configureAuth

The current implementation assumes that we already have docker running.
 This switches it to not remove any previous docker configuration
(since there isn't any), and uses our native file transfer utils
instead of the printf commands.

Also, the CopyLocal function can be deleted and the `bootstrapper.ExecRunner.Copy` can be used.  Its included since the CommandRunner isn't merged yet.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -95,7 +95,7 @@ out/localkube: $(shell $(LOCALKUBEFILES))
 ifeq ($(LOCALKUBE_BUILD_IN_DOCKER),y)
 	$(KUBE_CROSS_DOCKER_CMD) make $@
 else
-	CGO_ENABLED=1 go build -tags static_build,netgo -ldflags=$(LOCALKUBE_LDFLAGS) -o $(BUILD_DIR)/localkube ./cmd/localkube
```
Comment (dlorenc) [inline Makefile]: Looks like this has your netgo removal commit as well, want to merge that then rebase this one?

File: pkg/util/utils.go [hunk 2]
```diff
@@ -205,3 +205,30 @@ func MaybeChownDirRecursiveToMinikubeUser(dir string) error {
 	}
 	return nil
 }
+
+func CopyFile(src, dst string) error {
```
Comment (dlorenc) [inline pkg/util/utils.go]: We don't have anything for copying already?

File: pkg/provision/buildroot.go [hunk 3]
```diff
@@ -170,3 +177,90 @@ func setRemoteAuthOptions(p provision.Provisioner) auth.Options {
...
+	authOptions := p.GetAuthOptions()
+	org := mcnutils.GetUsername() + "." + machineName
+	bits := 2048
+
+	ip, err := driver.GetIP()
+	if err != nil {
+		return errors.Wrap(err, "error getting ip during provisioning")
+	}
+
+	if err := util.CopyFile(authOptions.CaCertPath, filepath.Join(authOptions.StorePath, "ca.pem")); err != nil {
```
Comment (dlorenc) [inline pkg/provision/buildroot.go]: Could you put these into a loop[ going over the map you have on 229 somehow?

Comment (r2d4) [inline pkg/provision/buildroot.go]: Oops, I think that was the point of the map down there anyways.  I'll change that

File: Makefile (hunk 1 above)
Comment (r2d4) [inline Makefile]: Yep that was the plan

File: pkg/util/utils.go (hunk 2 above)
Comment (r2d4) [inline pkg/util/utils.go]: Actually we have CopyFileLocal.  I'll use that.  This was just cherry-picked from an old branch

**Comment by r2d4** [comment]
@minikube-bot retest this please

**Pull request merged**